source/common/stream_info/stream_info_impl.h

```diff
@@ -100,6 +100,14 @@ struct StreamInfoImpl : public StreamInfo {
 
   absl::optional<uint32_t> responseCode() const override { return response_code_; }
 
+  const absl::optional<std::string>& responseCodeDetails() const override {
+    return response_code_details_;
+  }
+
+  void setResponseCodeDetails(absl::string_view rc_details) override {
```

Comment:
Is it up to the caller on how to enrich this information? Or do you imagine codifying this somehow, i.e a comma delimited list of details?

Reply:
My thinking was that this will be just one of the string constants defined in [ResponseCodeDetailValues](https://github.com/envoyproxy/envoy/blob/61928bdd0df1f4053a03548c99e17d7d404d2088/include/envoy/stream_info/stream_info.h#L76). That currently only has 1 value but that will change soon (#6542). Also, since this is a string, 3rd party extensions can add additional values if it doesn't make sense to add it to OSS envoy. I don't see a need for making this an array (or comma-separated list). @alyssawilk wdyt?

Reply:
Just to add another point - I think the idea is that there should be enough distinct strings available to uniquely specify why envoy is returning an error code (more or less one for each time sendLocalReply is called). So a list of strings should not be necessary.

Reply:
I'm fine with it as-is.

For single-hop responses we only need one rc-details.  It might be useful to have a full trace of what happened somewhere (500-retried, timeout-retries, gave up and sent 500) but I think that's better off via some other trace mechanism where we can put arbitrary information about how much time was spent in each filter etc. etc.

For multi-hop responses while we might want both layers captured in access logs (e.g. layer-2 envoy had timeout, layer-1 envoy proxied response code sent by upstream) I think that the information from prior hops would be transmitted via headers, and could be sent to access logs via logging the proxy-status header.  SG?

Reply:
Sounds good, thanks!